drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -492,6 +492,58 @@ static int rtc_stm32_init(const struct device *dev)
 	return 0;
 }
 
+int rtc_stm32_set_value(const struct device *dev, uint32_t ticks)
+{
+#if !defined(COUNTER_NO_DATE)
```

Comment:
If you reorder a bit, you can make sure that you only need 1 `#if !defined(COUNTER_NO_DATE)` block

Reply:
Actually 2 if `if (LL_RTC_DATE_Init(RTC, LL_RTC_FORMAT_BIN, &rtc_date) != SUCCESS) {` needs its own block

Reply:
`rtc_stm32_set_alarm`